rye/src/cli/add.rs

```diff
@@ -21,12 +21,16 @@ from unearth.finder import PackageFinder
 
 py_ver = tuple(map(int, sys.argv[1].split('.')))
 package = sys.argv[2]
+if len(sys.argv) > 3 and sys.argv[3] == "--pre=True":
+    allow_prereleases = True
+else:
+    allow_prereleases = False
 
 finder = PackageFinder(
     index_urls=["https://pypi.org/simple/"],
 )
 finder.target_python.py_ver = py_ver
-m = finder.find_matches(package)
+m = finder.find_matches(package, allow_prereleases=allow_prereleases)
```

Comment:
Originally I used `packaging` but I didn't want to add that to the boostrapping for `rye`. Ideally, this should work but it't not yet.